core/test/event_test.cpp

```diff
@@ -184,3 +186,179 @@ TEST_F(EventBroadcastTest, data_array_fail) {
   DataArray a(dense);
   EXPECT_THROW(event::broadcast_weights(a), except::EventDataError);
 }
+
+static auto make_sparse() {
+  auto var = makeVariable<event_list<double>>(
+      Dims{Dim::Z, Dim::Y}, units::Unit(units::us), Shape{3, 2});
+  scipp::index count = 0;
+  for (auto &v : var.values<event_list<double>>())
+    v.resize(count++);
+  return var;
+}
+
+static auto make_sparse_with_variances() {
+  auto var = makeVariable<event_list<double>>(
+      Dimensions{{Dim::Z, 3}, {Dim::Y, 2}}, Values{}, Variances{});
+  scipp::index count = 0;
+  for (auto &v : var.values<event_list<double>>())
+    v.resize(count++);
+  count = 0;
+  for (auto &v : var.variances<event_list<double>>())
+    v.resize(count++);
+  return var;
+}
+
+TEST(EventSizesTest, fail_dense) {
+  auto bad = makeVariable<double>(Values{1.0});
+  EXPECT_ANY_THROW(event::sizes(bad));
+}
+
+TEST(EventSizesTest, no_variances) {
+  const auto var = make_sparse();
+  auto expected = makeVariable<scipp::index>(Dims{Dim::Z, Dim::Y}, Shape{3, 2},
+                                             Values{0, 1, 2, 3, 4, 5});
+  EXPECT_EQ(event::sizes(var), expected);
+}
+
+TEST(EventSizesTest, variances) {
+  const auto var = make_sparse_with_variances();
+  auto expected = makeVariable<scipp::index>(Dims{Dim::Z, Dim::Y}, Shape{3, 2},
+                                             Values{0, 1, 2, 3, 4, 5});
+  EXPECT_EQ(event::sizes(var), expected);
+}
+
+struct EventFilterTest : public ::testing::Test {
+  Variable data_with_variances = makeVariable<event_list<float>>(
```

Comment:
Why the choice of `float` here?
I know it's probably part of the broader clenaup of tests task we have, but should these be made into a fixture that tests all the different types? (`float, double, int32, int64`)

Reply:
No reason in particular.

I don't think it is productive to test for all types on the top level in general. Ideally most of this is tested on lower levels, such as `transform` or the functors in the element-wise operations. There are still cases where we need explicit tests with various types for high-level code, but in general we should be able to rely on the underlying building blocks.

Reply:
I should actually add the element-functor tests for `copy_if` and `make_select`. If you don't mind I will do that in the follow-up #1048 since I added similar tests there already.